@here/harp-mapview/lib/Utils.ts

```diff
@@ -239,11 +258,34 @@ export namespace MapViewUtils {
             // previous computation to get the projection of the camera on the ground, then add
             // the height of the camera in the tangent space.
             const height = distance * Math.cos(pitchRad);
-            cache.vector3[1].add(tangentSpace.y).add(tangentSpace.z.setLength(height));
+            result.add(tangentSpace.y).add(tangentSpace.z.setLength(height));
+
+            const a = EarthConstants.EQUATORIAL_RADIUS + altitude;
+            const b = Math.sin(pitchRad) * distance;
+            const cameraHeight = Math.sqrt(a * a + b * b);
+            result.setLength(cameraHeight);
         }
 
-        // Convert back to GeoCoordinates and return result.
-        return mapView.projection.unprojectPoint(cache.vector3[1]);
+        return result;
+    }
+
+    export function getCameraCoordinatesFromTargetCoordinates(
+        targetCoordinates: GeoCoordinates,
+        distance: number,
+        yawDeg: number,
+        pitchDeg: number,
+        mapView: MapView
+    ): GeoCoordinates {
+        return mapView.projection.unprojectPoint(
+            getCameraPositionFromTargetCoordinates(
```

Comment:
:v: 

Do we actually need this method now?

Reply:
I left it b/c it is public API. I will mark it as deprecated though